js-client/src/nix-search.ts

```diff
@@ -0,0 +1,109 @@
+import { LitElement, css, html } from "lit";
+import { customElement, state } from "lit/decorators.js";
+import "@shoelace-style/shoelace/dist/themes/light.css";
+import "@shoelace-style/shoelace/dist/components/button/button.js";
+import "@shoelace-style/shoelace/dist/components/menu/menu.js";
+import "@shoelace-style/shoelace/dist/components/menu-item/menu-item.js";
+import "@shoelace-style/shoelace/dist/components/input/input.js";
+import "@shoelace-style/shoelace/dist/components/spinner/spinner.js";
+import "@shoelace-style/shoelace/dist/components/alert/alert.js";
+import "@shoelace-style/shoelace/dist/components/icon/icon.js";
+
+interface Operation {
+  type: "search" | "graph";
+  data: string[];
+}
+
+type EventInput = Event & { target: HTMLInputElement };
+
+@customElement("nix-search")
+export class NixSearch extends LitElement {
+  worker = new Worker(new URL("./packages", import.meta.url), {
+    type: "module",
+  });
+
+  @state()
+  pkgs: string[] = [];
+
+  @state()
+  loading = true;
+
+  constructor() {
+    super();
+    this.worker.addEventListener("message", (msg: MessageEvent<Operation>) => {
```

Comment:
What's the garantee that we have `msg: MessageEvent<Operation>`?

https://github.com/tweag/nixpkgs-graph-explorer/pull/37/files#diff-8a6044e9823d1a782d9032e7b7d939d39de15bf0278f0a1d723492d43cad04aeR92

does not seem typed at all

Reply:
I picked poor names. Those events are not related, I'm changing the name of the custom event to avoid confusions.